File: gzoom2-fe/app/src/app/api/model/partyRoleEx.ts
import { Party, PartyParentRole } from "./party";
import { PartyRole } from "./partyRole";
import { RoleType } from "./role-type";

export class PartyRoleEx extends PartyRole {
    public roleType?: RoleType;
    public party?: Party;
    public partyParentRole?: PartyParentRole;
}

File: gzoom2-be/persistence-mybatis/src/main/java/it/mapsgroup/gzoom/mybatis/dto/PartyRoleEx.java
package it.mapsgroup.gzoom.mybatis.dto;

public class PartyRoleEx extends PartyRole {

    private Party party;
    private PartyParentRole partyParentRole;
    private RoleType roleType;

    public Party getParty() {
        return party;
    }

    public void setParty(Party party) {
        this.party = party;
    }

    public PartyParentRole getPartyParentRole() {
        return partyParentRole;
    }

    public void setPartyParentRole(PartyParentRole partyParentRole) {
        this.partyParentRole = partyParentRole;
    }

    public RoleType getRoleType() {
        return roleType;
    }

    public void setRoleType(RoleType roleType) {
        this.roleType = roleType;
    }
}
